Add check to run go generate ./... after vendor changes

gargnupur opened the pull request:
This is same as https://github.com/istio/istio/pull/6345/

istio-testing commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/istio/istio/pull/6606#" title="Author self-approved">gargnupur</a>*
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **geeknoid**

Assign the PR to them by writing `/assign @geeknoid` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/istio/istio/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["geeknoid"]} -->

istio-testing commented:
@gargnupur: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-unit-tests.sh | 664618c2f5f9e96a5bb9144bbe3d4efd9d79f6f7 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/6606/istio-unit-tests/9298/) | `/test istio-unit-tests`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

ozevren commented:
It looks like there is quite a few somewhat independent changes in the PR. Would it make sense to split them and send that way? For example, the fixes to fakes seems completely separable. Similarly, the check for go generate can be separated from running the go generate (and getting the repo state upto speed) itself.

ozevren commented on bin/install_protoc.sh:
```diff
@@ -0,0 +1,54 @@
+        echo "unknown: ${OSTYPE}"
+        exit 1;;
+    esac
+    sudo mv protobuf-all/bin/* /usr/local/bin
```
This doesn't look right to me. Please do not modify the developer's machine state.

gargnupur replied:
I do this only if developer does not already have protoc installed and developer is suggested to run depend.diff only if they have vendor changes. In that case I think this is a help to developer so that all the required binaries for go generate are installed for them, if they don't have it already...

Also, we do install stuff for things like linters: 
https://github.com/istio/istio/blob/master/bin/linters.sh#L29:7

ozevren replied:
Modifying /usr/.../bin is a big no-no. In general, it is a big source of subtle issues for the developer, in addition to being a potential security problem. If nothing else, the assumption that the machine is being used for Istio development only is wrong. (It is quite likely that it is used to develop multiple projects which may have different versioned tooling) From this regard, go get has relatively reduced scope (which is easy to isolate) and does not require privileges. 

Re: linters We actually used to do docker based linters. (I am not sure why we're not doing it anymore). Api repo uses docker images as well. 

It is better to document the prerequisites, and loudly complain when they are not found.

gargnupur replied:
Thanks.. that makes sense.. how about adding a big red warning in red and going ahead and installing only if user says yes..

Is there a way to check that script is running in circle ci? So that we don't wait for user input when running this script later in circle ci?

ozevren replied:
We can document the prerequisite here:
https://github.com/istio/istio/wiki/Dev-Guide

Also, why do we need protoc 3.5.1 explicitly? Do you need to peg the version number to get consistent diffs? 

A cleaner approach is probably to update all go:generates that makes direct calls to protoc to be updated to use one from a docker image. The API repo already uses a protoc in a docker image, so you can simply use it. To make things simple, you can create a "protoc.sh" in bin/, that acts just like regular protoc, but uses docker run internally (i.e. it simply passes its command-line flags to docker run). 
https://github.com/istio/api/blob/master/Makefile

mandarjog replied:
pinned protoc version is indeed needed to get a consistent diff.

gargnupur replied:
@ozevren : what is the version used here? https://github.com/istio/api/blob/master/Makefile#L14:32

gargnupur commented:
@ozevren : Thanks for the feedback. 
Removed files generated by go generate from PR. Will send them in a separate PR.
Also, not running go generate for user currently and added it as a separate option just for check under go.generate.diff

istio-testing commented:
@gargnupur: PR needs rebase.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

istio-testing commented:
@gargnupur: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/e2e-bookInfoTests-v1alpha3.sh | 0d506568dc1f72430ee978b8d99b5453fe72aaf2 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/6606/e2e-bookInfoTests-envoyv2-v1alpha3/3175/) | `/test e2e-bookInfo-envoyv2-v1alpha3`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

ozevren commented:
I don't think it is using a fixed version. The docker image is being
generated from here:
https://github.com/istio/api/blob/master/docker/Dockerfile#L30


On Fri, Jul 13, 2018 at 3:36 PM gargnupur <notifications@github.com> wrote:

> *@gargnupur* commented on this pull request.
> ------------------------------
>
> In bin/install_protoc.sh
> <https://github.com/istio/istio/pull/6606#discussion_r202488485>:
>
> > +        curl -OL https://github.com/google/protobuf/releases/download/v3.5.1/protoc-3.5.1-osx-$LOCAL_ARCH.zip
> +        unzip protoc-3.5.1-osx-$LOCAL_ARCH.zip -d protobuf-all
> +        rm -f protoc-3.5.1-osx-$LOCAL_ARCH.zip ;;
> +      linux*)
> +        curl -OL https://github.com/google/protobuf/releases/download/v3.5.1/protoc-3.5.1-linux-$LOCAL_ARCH.zip
> +        unzip protoc-3.5.1-linux-$LOCAL_ARCH.zip -d protobuf-all
> +        rm -f protoc-3.5.1-linux-$LOCAL_ARCH.zip;;
> +      msys)
> +        curl -OL https://github.com/google/protobuf/releases/download/v3.5.1/protoc-3.5.1-win32.zip
> +        unzip protoc-3.5.1-win32.zip -d protobuf-all
> +        rm -f protoc-3.5.1-win32.zip ;;
> +      *)
> +        echo "unknown: ${OSTYPE}"
> +        exit 1;;
> +    esac
> +    sudo mv protobuf-all/bin/* /usr/local/bin
>
> @ozevren <https://github.com/ozevren> : what is the version used here?
> https://github.com/istio/api/blob/master/Makefile#L14:32
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/istio/istio/pull/6606#discussion_r202488485>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AQR8I8ryCEEgkDVvBajiLK0PUaQnE7hRks5uGSEJgaJpZM4U26jP>
> .
>

istio-testing commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/istio/istio/pull/6606#" title="Author self-approved">gargnupur</a>*
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **geeknoid**

If they are not already assigned, you can assign the PR to them by writing `/assign @geeknoid` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/istio/istio/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["geeknoid"]} -->

istio-testing commented:
@gargnupur: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/e2e-bookInfoTests-v1alpha3.sh | 0d506568dc1f72430ee978b8d99b5453fe72aaf2 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/6606/e2e-bookInfoTests-envoyv2-v1alpha3/3175/) | `/test e2e-bookInfo-envoyv2-v1alpha3`
prow/e2e-simpleTests.sh | 63ce6ef842026abd52e03b700ad4ffb0b905d4d8 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/6606/e2e-simpleTests/6681/) | `/test e2e-simple`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **geeknoid**

If they are not already assigned, you can assign the PR to them by writing `/assign @geeknoid` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/istio/istio/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["geeknoid"]} -->

istio-testing commented:
@gargnupur: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/e2e-mixer-no_auth.sh | 578a529a8d916253f1d127ba6202307f8e735c5b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/6606/e2e-mixer-no_auth/2302/) | `/test e2e-mixer-no_auth`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

gargnupur commented:
/test e2e-mixer-no_auth

gargnupur commented:
/test e2e-pilot-noauth-v1alpha3-v2

gargnupur commented:
/retest

istio-testing commented:
@gargnupur: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-unit-tests.sh | 578a529a8d916253f1d127ba6202307f8e735c5b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/6606/istio-unit-tests/11402/) | `/test istio-unit-tests`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@gargnupur: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-unit-tests.sh | 578a529a8d916253f1d127ba6202307f8e735c5b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/6606/istio-unit-tests/11402/) | `/test istio-unit-tests`
prow/e2e-mixer-no_auth.sh | 578a529a8d916253f1d127ba6202307f8e735c5b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/6606/e2e-mixer-no_auth/2327/) | `/test e2e-mixer-no_auth`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@gargnupur: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/e2e-mixer-no_auth.sh | 578a529a8d916253f1d127ba6202307f8e735c5b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/6606/e2e-mixer-no_auth/2327/) | `/test e2e-mixer-no_auth`
prow/istio-unit-tests.sh | f70419fb1039460661647dc8e4f44f0e6dfef5f9 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/6606/istio-unit-tests/11456/) | `/test istio-unit-tests`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@gargnupur: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/e2e-mixer-no_auth.sh | 578a529a8d916253f1d127ba6202307f8e735c5b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/6606/e2e-mixer-no_auth/2327/) | `/test e2e-mixer-no_auth`
prow/istio-unit-tests.sh | f70419fb1039460661647dc8e4f44f0e6dfef5f9 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/6606/istio-unit-tests/11456/) | `/test istio-unit-tests`
prow/e2e-bookInfoTests-v1alpha3.sh | f70419fb1039460661647dc8e4f44f0e6dfef5f9 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/6606/e2e-bookInfoTests-envoyv2-v1alpha3/4548/) | `/test e2e-bookInfo-envoyv2-v1alpha3`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@gargnupur: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-unit-tests.sh | f70419fb1039460661647dc8e4f44f0e6dfef5f9 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/6606/istio-unit-tests/11456/) | `/test istio-unit-tests`
prow/e2e-bookInfoTests-v1alpha3.sh | f70419fb1039460661647dc8e4f44f0e6dfef5f9 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/6606/e2e-bookInfoTests-envoyv2-v1alpha3/4548/) | `/test e2e-bookInfo-envoyv2-v1alpha3`
prow/istio-pilot-multicluster-e2e.sh | f70419fb1039460661647dc8e4f44f0e6dfef5f9 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/6606/istio-pilot-multicluster-e2e/386/) | `/test istio-pilot-multicluster-e2e`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@gargnupur: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-pilot-multicluster-e2e.sh | 1b2157b12eb84530566e4e2dbe10937ac36b14b7 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/6606/istio-pilot-multicluster-e2e/419/) | `/test istio-pilot-multicluster-e2e`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

gargnupur commented:
@geeknoid , @ozevren : Can you please review this cl? It adds a rule in makefile to run go generate ./...
 and a rule in circleci to do the same..

geeknoid commented:
/lgtm
/approve

istio-testing commented:
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/istio/istio/pull/6606#issuecomment-413374495" title="Approved">geeknoid</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/istio/istio/blob/master/OWNERS)~~ [geeknoid]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->

istio-testing commented:
@gargnupur: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/e2e-simpleTests.sh | 047e37a544e5acf2dd39766825e99afbbb8a47e6 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/6606/e2e-simpleTests/7701/) | `/test e2e-simple`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

gargnupur commented:
/test e2e-simple

istio-testing commented:
@gargnupur: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/e2e-simpleTests.sh | 047e37a544e5acf2dd39766825e99afbbb8a47e6 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/6606/e2e-simpleTests/7752/) | `/test e2e-simple`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

gargnupur commented:
/test e2e-simple

istio-testing commented:
@gargnupur: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/e2e-simpleTests.sh | 047e37a544e5acf2dd39766825e99afbbb8a47e6 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/6606/e2e-simpleTests/7753/) | `/test e2e-simple`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

geeknoid commented:
/test e2e-simple

istio-testing commented:
@gargnupur: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/e2e-simpleTests.sh | 047e37a544e5acf2dd39766825e99afbbb8a47e6 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/6606/e2e-simpleTests/7755/) | `/test e2e-simple`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

gargnupur commented:
e2e-simple test is failing because citadel pod is not crashing and from logs, error is "2018-08-16T17:11:10.036080Z	error	unknown flag: --identity-domain" -> this can't be coming from my pr..

will try retest again and if that failed too.. will rebase the pr again..

gargnupur commented:
/test e2e-simple

gargnupur commented:
/test e2e-simple

istio-testing commented:
@gargnupur: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/e2e-simpleTests.sh | 047e37a544e5acf2dd39766825e99afbbb8a47e6 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/6606/e2e-simpleTests/7795/) | `/test e2e-simple`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
New changes are detected. LGTM label has been removed.

istio-testing commented:
@gargnupur: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/e2e-simpleTests.sh | 047e37a544e5acf2dd39766825e99afbbb8a47e6 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/6606/e2e-simpleTests/7795/) | `/test e2e-simple`
prow/istio-unit-tests.sh | 65f25912887e2eb0513f9c4574628f5decfa175b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/6606/istio-unit-tests/11746/) | `/test istio-unit-tests`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

geeknoid commented:
/lgtm
/approve

istio-testing commented:
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/istio/istio/pull/6606#issuecomment-413925700" title="Approved">geeknoid</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/istio/istio/blob/master/OWNERS)~~ [geeknoid]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->

istio-testing commented:
@gargnupur: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-unit-tests.sh | 65f25912887e2eb0513f9c4574628f5decfa175b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/6606/istio-unit-tests/11746/) | `/test istio-unit-tests`
prow/e2e-bookInfoTests-v1alpha3.sh | 65f25912887e2eb0513f9c4574628f5decfa175b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/6606/e2e-bookInfoTests-envoyv2-v1alpha3/4833/) | `/test e2e-bookInfo-envoyv2-v1alpha3`
prow/e2e-mixer-no_auth.sh | 65f25912887e2eb0513f9c4574628f5decfa175b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/6606/e2e-mixer-no_auth/2651/) | `/test e2e-mixer-no_auth`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
New changes are detected. LGTM label has been removed.

istio-testing commented:
@gargnupur: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/e2e-bookInfoTests-v1alpha3.sh | 65f25912887e2eb0513f9c4574628f5decfa175b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/6606/e2e-bookInfoTests-envoyv2-v1alpha3/4833/) | `/test e2e-bookInfo-envoyv2-v1alpha3`
prow/e2e-mixer-no_auth.sh | 65f25912887e2eb0513f9c4574628f5decfa175b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/6606/e2e-mixer-no_auth/2651/) | `/test e2e-mixer-no_auth`
prow/istio-unit-tests.sh | d9be1bc81c79e3aba25e1c630f1b35e2104c8c91 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/6606/istio-unit-tests/11859/) | `/test istio-unit-tests`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@gargnupur: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-pilot-multicluster-e2e.sh | 9f63815bbd0ccf6d9ca5cbbb2cc03bf7443d4299 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/6606/istio-pilot-multicluster-e2e/793/) | `/test istio-pilot-multicluster-e2e`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->
